### ISSUE TITLE
AnomalyScore and multiStepPredictions.N in OPF do not have correct (strict) data type (can be None)

### ISSUE THREAD

**Issue by breznak** [numenta/nupic#2802]
* `anomalyScore` is defined as a float in range [0..1], while in OPF results file it has type "string" (because a few 1st values can be `None`)
* that is because `(multistep)prediction models` predict the 1st N values as `None`, instead of desired data type again. 

This could be easily fixed, anomalyScore should be `0.5` during burn-in, predictions could be eg `0.0`. 
That is a more correct approach, supports strict typing, and some applications could be easier. 

Our simple use-case is in https://github.com/nupic-community/nupic.visualizations/issues/30

**Comment by rhyolight** [comment]
Is it currently an empty string to denote no value?

**Comment by breznak** [comment]
anomalyScore:
 * type: string (incorrect)
 * default value: `0.5` (OK)

predictions:
  * string (in numeric cases incorrect, should allow to set strongly typed value)
  * `None`

**Comment by rhyolight** [comment]
@subutai @scottpurdy I tend to agree with @breznak here.

**Comment by subutai** [comment]
Sounds good to me too. It also matches what we do with AnomalyLikelihood.

**Comment by scottpurdy** [comment]
I agree that the anomaly score should be a float and it should match what is returned [here](https://github.com/numenta/nupic/blob/299e0557c15c2233b71ad72811f4d375e4324ae7/src/nupic/algorithms/anomaly.py#L30-L52) rather than whatever manipulation leads to a string or None. We definitely should not set it to `0.5` arbitrarily.

The predictions can be empty though and it definitely does not make sense to have a default value. `0.0` is no more accurate than `42` when you haven't learned anything yet. When the predictions are returned as a dict this shouldn't be an issue. For the multiStepBestPrediction, `None` seems the best value to me.

Keep in mind that changing from string to int or float could be a complicated change to implement. Hopeful not but who knows.
